### PR TITLE
Alternative Rector workflow on check repo before steps run

### PR DESCRIPTION
**Pull request by samsonasik**
@TomasVotruba this is alternative PR of:

- https://github.com/rectorphp/rector-src/pull/3439

which move the check on before steps instead of before strategy.

Closes https://github.com/rectorphp/rector-src/pull/3439

### CODE REVIEW

File: .github/workflows/rector.yaml [hunk 1]
```diff
@@ -19,20 +16,15 @@ jobs:
...
                 uses: actions/checkout@v3
                 with:
                     # Must be used to trigger workflow after push
                     token: ${{ secrets.ACCESS_TOKEN }}
 
-            # in forks, the token is not available - so we cannot use it
-            -
-                if: github.event.pull_request.head.repo.full_name != github.repository
-                uses: actions/checkout@v3
+            - run: echo "run on ${{ github.event.pull_request.head.repo.full_name }}"
```
Comment (samsonasik) [inline .github/workflows/rector.yaml:27]: @TomasVotruba this is to verify the repo name to verify and it show:

```
Run echo "run on rectorphp/rector-src"
run on rectorphp/rector-src
```

see https://github.com/rectorphp/rector-src/actions/runs/4316268191/jobs/7531815857#step:3:1

Comment (samsonasik) [reply .github/workflows/rector.yaml:27]: The `pull_request.head` seems goes to forks repo name
while `github.repository` goes to target repo name, so I changed to `github.event.pull_request.head.repo.full_name`

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba  let's give it a try? 

@jackbentley you need to rebase your PR https://github.com/rectorphp/rector-src/pull/3430 again to verify if it keep works after this merged.

**Comment by TomasVotruba** [comment]
Let's try it out 🙏

**Comment by samsonasik** [comment]
@TomasVotruba it doesn't work, it now show pending on PR from fork, see https://github.com/rectorphp/rector-src/pull/3430#issuecomment-1452371965

**Pull request merged**